modules/post/linux/manage/xdg_screensaver.rb

```diff
@@ -0,0 +1,48 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Linux::System
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Linux XDG screensaver',
+      'Description'   => %q{
+          This module will allows you to control the screensaver of the target system.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => ['DeveloppSoft'],
+      'Platform'      => ['linux'],
+      'SessionTypes'  => ['shell', 'meterpreter'],
+      'Actions'       =>
+        [
+          ['STATUS',   { 'Description' => 'Print if the screensaver is enabled to turn on after a period of inactivity' } ],
+          ['LOCK',     { 'Description' => 'Lock current session' } ],
+          ['ACTIVATE', { 'Description' => 'Start the screensaver, session might be locked' } ],
+          ['RESET',    { 'Description' => 'Desactivate screensaver, user might have to unlock its session'}]
+        ],
+      'DefaultAction' => 'STATUS'
+    ))
+  end
+
+  def run
+    if action.nil?
+      print_error("Please specify an action")
+      return
+    end
+
+    case action.name
+      when 'STATUS'
```

Comment:
That's up to you. I figured it would be overkill and slower to check for each `ACTION`, although more user-friendly.

It makes sense to at least perform the check in the `STATUS` option, especially as it's the default.

Reply:
Made it only for status, feels like it is enough indeed!

Reply:
If you do decide to put the check for all actions, it would make more sense to put it before the `case` block; rather than put the `case` inside the `unless` block.

Reply:
this also might be better as a multi module from the start:

https://stackoverflow.com/questions/1430108/how-to-turn-screensaver-on-windows-7-by-a-code-in-cmd
https://stackoverflow.com/questions/4505198/how-can-i-start-the-screensaver-and-lock-the-screen-from-the-os-x-terminal

Reply:
Will also look at that